request.go

```diff
@@ -31,6 +31,10 @@ const (
 	HeaderRequestCharge       = "x-ms-request-charge"
 	HeaderAIM                 = "A-IM"
 	HeaderPartitionKeyRangeID = "x-ms-documentdb-partitionkeyrangeid"
+	HeaderUserAgent           = "User-Agent"
+
+	ClientName    = "documentdb-go"
+	ClientVersion = "1.3.0"
```

Comment:
Do we really need this information? If so, why not extracting it from the [BuildInfo](https://pkg.go.dev/runtime/debug#BuildInfo) struct? I just prefer to avoid maintaining it manually.

Reply:
Extracting it from the BuildInfo is the correct way, I'll go with that.

The client version in the user agent helps with debugging whether you've got an outdated client running somewhere. Useful if there's some bug you want to quickly check applies or not. I see a lot of users that catch zombie nodes after noticing an outdated client version in a user agent.